test/unit/index.js

```diff
@@ -155,47 +161,121 @@ describe('extractData()', function () {
                 expect(extractData).withArgs({locales: ['foo-bar']}).to.throwException();
             });
 
+            it('should always contribute an entry for all specified `locales`', function () {
+                var data = extractData({
+                    locales       : ['en-US', 'zh-Hans-SG'],
+                    pluralRules   : true,
+                    relativeFields: true,
+                });
+
+                expect(data).to.have.keys('en-US', 'zh-Hans-SG');
+            });
+
             it('should recursively expand `locales` to their roots', function () {
                 var data = extractData({
-                    locales       : ['en-US', 'zh-Hant-TW'],
+                    locales       : ['en-US', 'zh-Hans-SG'],
```

Comment:
Updated this to a locale hierarchy that doesn't hit an exceptional case — those are handled later in the tests.